lib/ansible/modules/system/java_keystore.py

```diff
@@ -0,0 +1,271 @@
+
+    return private_key_path
+
+
+def cert_changed(module, keystore_path, keystore_pass, alias):
+    certificate_path = create_tmp_certificate(module)
+    try:
+        current_certificate_fingerprint = read_certificate_fingerprint(module, certificate_path)
+        stored_certificate_fingerprint = read_stored_certificate_fingerprint(module, alias, keystore_path, keystore_pass)
+        return current_certificate_fingerprint != stored_certificate_fingerprint
+    finally:
+        os.remove(certificate_path)
+
+
+def create_jks(module, name, keystore_path, password):
+    if module.check_mode:
+        module.exit_json(changed=True)
+    else:
+        certificate_path = create_tmp_certificate(module)
+        private_key_path = create_tmp_private_key(module)
+        try:
+            if os.path.exists(keystore_path):
+                os.remove(keystore_path)
+
+            keystore_p12_path = "/tmp/keystore.p12"
+            if os.path.exists(keystore_p12_path):
+                os.remove(keystore_p12_path)
+
+            export_p12_cmd = "openssl pkcs12 -export -name '%s' -in '%s' -inkey '%s' -out '%s' -passout 'pass:%s'" % (
+                name, certificate_path, private_key_path, keystore_p12_path, password)
+            (rc, export_p12_out, export_p12_err) = module.run_command(export_p12_cmd, check_rc=True)
+            if rc != 0:
+                return module.fail_json(msg=export_p12_out, rc=rc, cmd=export_p12_out)
+
+            import_keystore_cmd = "keytool -importkeystore " \
+                                  "-destkeystore '%s' " \
+                                  "-srckeystore '%s' " \
+                                  "-srcstoretype pkcs12 " \
+                                  "-alias '%s' " \
+                                  "-deststorepass '%s' " \
+                                  "-srcstorepass '%s' " \
+                                  "-noprompt" % (keystore_path, keystore_p12_path, name, password, password)
+            (rc, import_keystore_out, import_keystore_err) = module.run_command(import_keystore_cmd, check_rc=True)
+            if rc == 0:
+                update_jks_perm(module, keystore_path)
+                return module.exit_json(changed=True, msg=import_keystore_out, rc=rc, cmd=import_keystore_cmd, stdout_lines=import_keystore_out)
+            else:
+                return module.fail_json(msg=import_keystore_out, rc=rc, cmd=import_keystore_cmd)
+        finally:
+            os.remove(certificate_path)
+            os.remove(private_key_path)
+
+
+def update_jks_perm(module, keystore_path):
+    module.params['path'] = keystore_path
+    file_args = module.load_file_common_arguments(module.params)
+    module.set_fs_attributes_if_different(file_args, False)
+
+
+def process_jks(module):
+    name = module.params['name']
+    password = module.params['password']
+    keystore_path = module.params['dest']
+    force = module.params['force']
+
+    test_keytool(module)
```

Comment:
Instead of this function, there is Ansible built-in function - `get_bin_path` which does lot of other things and tested.